### ISSUE TITLE
[Bug] Resource/groups/activity URLs doesn't support schema name.

### ISSUE THREAD

**Issue by thenets** [ckan/ckan#3603]
### CKAN Version / site URL
CKAN 2.6.2
http://dados.gov.br

I've created a issue about it in the ckanext-scheming GitHub page:
https://github.com/ckan/ckanext-scheming/issues/157
I don't know how the "IDatasetForm" works, but if anyone help me I can try to fix it in the next CKAN version.

## Problem
The scheming doesn't support the actions 'resource', 'groups', 'activity'.

These URL works ok:
http://localhost:5000/concurso/concurso-1
http://localhost:5000/dataset/concurso-1/resource/9a31c1ca-4b68-48c5-a1f1-668e5e5f01b1
http://localhost:5000/dataset/groups/concurso-1
http://localhost:5000/dataset/activity/concurso-1

But these URLs return error 404:
http://localhost:5000/concurso/concurso-1/resource/9a31c1ca-4b68-48c5-a1f1-668e5e5f01b1
http://localhost:5000/concurso/groups/concurso-1
http://localhost:5000/concurso/activity/concurso-1

**Comment by amercader** [comment]
@thenets thanks for the excellent report. This is something that should be easy to fix. Here is where the custom routes for each package type are registered:

https://github.com/ckan/ckan/blob/0bb141ae43b51f2c095fba87750c5fdaa00ee816/ckan/lib/plugins.py#L93:L122

The easy fix would be to add the missing routes here, eg

```python
map.connect('%s_resource_read' % package_type, '/%s/{id}/resource/{resource_id}' % package_type,
    controller='package', action='resource_read')
```

But in fact I can't see a reason why we could not factor out the code the code on [routing.py](https://github.com/ckan/ckan/blob/0bb141ae43b51f2c095fba87750c5fdaa00ee816/ckan/config/routing.py#L211:L282) that registers *all* dataset routes to a function and call it with the dataset type, something like:

```python

# plugins.py

    for plugin in plugins.PluginImplementations(plugins.IDatasetForm):
        # ...
        for package_type in plugin.package_types():
            register_package_routes(map, package_type)


# routing.py

def make_map():
     # ...

    register_package_routes(map)


def register_package_routes(map, package_type='dataset'):
     # Same route registration as currently, but using the relevant package_type (`dataset` or a custom one)

     map.connect('%s_search' % package_type, '/%s' % package_type,
                        controller='package', action='search')

```

@wardi you know more `IDatasetForm`, can you see a problem with this approach?

@thenets do you want to have a go at submitting a patch for this? either the easy or the better option, I can help if necessary.

**Comment by wardi** [comment]
@amercader I don't see a problem with that

**Comment by thenets** [comment]
@amercader Thanks! I'll fix that in the Brazillian plugin!

About the CKAN core, I don't know a lot about Python, but I'll try to create a fix to that and test in my CKAN 2.6.x.
Later I sent a pull request and you and/or @wardi help me to test on 2.7.x.

Thanks for the tips about CKAN routing!!!

**Comment by kmbn** [comment]
We decided to close old issues that are not actively worked on so that we can focus our effort and attention on issues affecting the current versions of CKAN.

If this issue is still affecting the version of CKAN you're working with now, please feel free to comment or reopen the issue.

If you do reopen this issue, please update it with new details. One reason it might not have been resolved in the past is that it wasn't clear how a contributor could address the issue.
